vega_sim/scenario/common/agents.py

```diff
@@ -1129,3 +1138,235 @@ def initialise(self, vega: VegaServiceNull):
+    """
+    Trading Agent that can follow multiple momentum trading strategies.
+
+    At each step, the trading agent collects future price and trades under
+    certain momentum indicator.
+    """
+
+    def __init__(
+        self,
+        wallet_name: str,
+        wallet_pass: str,
+        market_name: str,
+        asset_name: str,
+        indicator_threshold: Tuple[float, float] = (
+            75,
+            25,
+        ),
+        momentum_strategy: Tuple[str, Dict[str, float]] = (
+            "MACD",
+            {
+                "fast_period": 12,
+                "slow_period": 26,
+                "signal_period": 9,
+            },
+        ),
+        initial_asset_mint: float = 1e5,
+        buy_intensity: float = 5,
+        sell_intensity: float = 5,
+        base_order_size: float = 1,
+        trading_proportion: float = 1,
+        random_state: Optional[np.random.RandomState] = None,
+        send_limit_order: bool = False,
+        offset_levels: int = 10,
+        tag: str = "",
+    ):
+        super().__init__(wallet_name, wallet_pass)
+        self.market_name = market_name
+        self.asset_name = asset_name
+        self.initial_asset_mint = initial_asset_mint
+        self.buy_intensity = buy_intensity
+        self.sell_intensity = sell_intensity
+        self.base_order_size = base_order_size
+        self.tag = tag
+        self.momentum_strategy = momentum_strategy
+        self.indicator_threshold = indicator_threshold
+        self.trading_proportion = trading_proportion
+        self.random_state = (
+            random_state if random_state is not None else np.random.RandomState()
+        )
+        self.send_limit_order = send_limit_order
+        self.offset_levels = offset_levels
+
+        self.prices = []
+
+        self.momentum_func_dict = {
+            "RSI": self._RSI,
+            "CMO": self._CMO,
+            "STOCHRSI": self._STOCHRSI,
+            "APO": self._APO,
+            "MACD": self._MACD,
+        }
+
+        self.indicators = []
+
+    def initialise(self, vega: VegaServiceNull):
+        super().initialise(vega=vega)
+
+        self.market_id = [
+            m.id
+            for m in self.vega.all_markets()
+            if m.tradable_instrument.instrument.name == self.market_name
+        ][0]
+
+        self.asset_id = self.vega.find_asset_id(symbol=self.asset_name)
+        self.vega.mint(
+            self.wallet_name,
+            asset=self.asset_id,
+            amount=self.initial_asset_mint,
+        )
+
+        self.pdp = self.vega._market_pos_decimals.get(self.market_id, {})
+        self.mdp = self.vega._market_price_decimals.get(self.market_id, {})
+        self.adp = self.vega._asset_decimals.get(self.asset_id, {})
+        self.vega.wait_for_total_catchup()
+
+    def step(self, vega_state: VegaState):
+        self._collect_prices()
+        signal = self.momentum_func_dict.get(self.momentum_strategy[0])()
```

Comment:
The `get` isn't too useful here since returning `None` would just result in attempting to call `None()` which would break anyway, but can leave this time.

Reply:
I added `self._RSI`in the get method, so avoid to call `None` if there is no momentum strategy corresponded.